ui/app/components/app/transaction-breakdown/transaction-breakdown.component.js

```diff
@@ -47,13 +48,18 @@ export default class TransactionBreakdown extends PureComponent {
             )
           }
         </TransactionBreakdownRow>
-        <TransactionBreakdownRow title={t('amount')}>
-          <UserPreferencedCurrencyDisplay
-            className="transaction-breakdown__value"
-            type={PRIMARY}
-            value={value}
-          />
-        </TransactionBreakdownRow>
+        {isTokenApprove
+          ? (
+            <TransactionBreakdownRow title={t('spendingLimitAmount')}>
+              <span className="transaction-breakdown__value">{primaryCurrency}</span>
+            </TransactionBreakdownRow>
+          )
+          : (
+            <TransactionBreakdownRow title={t('amount')}>
+              <span className="transaction-breakdown__value">{primaryCurrency}</span>
+            </TransactionBreakdownRow>
+          )
+        }
```

Comment:
Minor suggestion on improving this:

```suggestion
        <TransactionBreakdownRow
          title={
            isTokenApprove
              ? t('spendingLimitAmount')
              : t('amount')
          }
        >
          <span className="transaction-breakdown__value">{primaryCurrency}</span>
        </TransactionBreakdownRow>
```